tools/testgrid-analysis/cmd/github.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/v60/github"
+)
+
+func createIssues(tests []*TestResultSummary, labels []string) {
+	openIssues := getOpenIssues(labels)
+	for _, t := range tests {
+		createIssueIfNonExist(tab, t, openIssues, append(labels, "help wanted"))
+	}
+}
+
+func getOpenIssues(labels []string) []*github.Issue {
+	client := github.NewClient(nil).WithAuthToken(os.Getenv("GITHUB_TOKEN"))
+	ctx := context.Background()
+	// list open issues with label type/flake
+	issueOpt := &github.IssueListByRepoOptions{Labels: labels}
+	issues, _, err := client.Issues.ListByRepo(ctx, githubOwner, githubRepo, issueOpt)
```

Comment:
Heads up that the GitHub API paginates the responses. The default number of items per page is 30 (https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#list-repository-issues). Ideally, there should be at most 30 open issues for flaky tests, but if there are, it may create duplicate issues.

Reply:
thanks for the pointer. Updated the logic to include pagination